kubernetes-extension/src/main/java/org/ballerinax/kubernetes/utils/KubernetesUtils.java

```diff
@@ -369,19 +392,32 @@ public static String getStringValue(BLangExpression expr) throws KubernetesPlugi
         }
         throw new KubernetesPluginException("unable to parse value: " + expr.toString());
     }
-
+    
     /**
-     * Generate set of string using a {@link BLangArrayLiteral}.
+     * Parse build extension of @kubernetes:Deployment annotation.
      *
-     * @param bArrayLiteral Array literal.
-     * @return Convert string.
+     * @param buildExtensionValue Fields of the buildExtension field.
+     * @return Build extension model.
+     * @throws KubernetesPluginException When an unknown extension is found.
      */
-    public static Set<String> getArray(BLangArrayLiteral bArrayLiteral) throws KubernetesPluginException {
-        Set<String> scopeSet = new LinkedHashSet<>();
-        for (ExpressionNode bLangExpression : bArrayLiteral.getExpressions()) {
-            scopeSet.add(resolveValue(bLangExpression.toString()));
+    public static DeploymentBuildExtension parseBuildExtension(BLangExpression buildExtensionValue)
+            throws KubernetesPluginException {
+        if (buildExtensionValue instanceof BLangSimpleVarRef || buildExtensionValue instanceof BLangLiteral) {
```

Comment:
Here without using 'instanceof' we can check the node kind.

Reply:
This cannot be done as I am expecting a parent type `BLangSimpleVarRef` or `BLangLiteral`.